pyuvdata/uvbeam.py

```diff
@@ -1102,54 +1156,56 @@ def _get_beam(self, pol):
         # assert type is int, not string
         if isinstance(pol, (str, np.str)):
             pol = uvutils.polstr2num(pol)
-        # get pol-string
-        polstr = uvutils.polnum2str(pol)
-        # pols dict
-        pols_dict = {1: (-5, -6), 2: (-5, -6), 3: (-7, -8), 4: (-7, -8)}
-        # get pol array
         pol_array = self.polarization_array
-        # pQ, pU and pV yet to be implemented
-        allowed = [1, -5, -6, -7, -8]
-        if pol in allowed:
+        if pol in [-5, -6, -7, -8]:  # linear polarizations
             if pol in pol_array:
                 stokes_p_ind = np.where(np.isin(pol_array, pol))[0][0]
                 beam = self.data_array[0, 0, stokes_p_ind]
-            elif pol == 1:
-                keys = pols_dict[1]
-                if keys[0] in pol_array and keys[1] in pol_array:
-                    xx_ind = np.where(np.isin(pol_array, keys[0]))[0][0]
-                    yy_ind = np.where(np.isin(pol_array, keys[1]))[0][0]
-                    beam = 0.5 * (self.data_array[0, 0, xx_ind] + self.data_array[0, 0, yy_ind])
-                else:
-                    raise ValueError('Do not have the right polarization information')
             else:
                 raise ValueError('Do not have the right polarization information')
+        elif pol in [1, 2, 3, 4]:  # pseudo stokes
+            assert self.beam_type == 'efield', "Beam type must be efield for pseudo stokes {}".format(uvutils.polnum2str(pol))
```

Comment:
The beam can in principal already have been converted to a power beam in pseudo stokes (e.g. using your conversion code)  then you'd have a power beam with the polarization array having values 1:4. In that case, I think you should be able just get the values from the beam.

Reply:
I've included a separate function efield_to_pstokes which does that and then call be called in _get_beam.